src/v/archival/ntp_archiver_service.cc

```diff
@@ -272,11 +272,23 @@ ss::future<cloud_storage::download_result> ntp_archiver::sync_manifest() {
     } else {
         vlog(
           _rtclog.debug, "Updating the archival_meta_stm in read-replica mode");
+        cloud_storage::partition_manifest mdiff(_ntp, _rev);
+        auto offset
```

Comment:
When last offset is {}, this sets `offset` to a value which will not match any segments in the newly downloaded manifest, and the read replica never sees any data.

I will push a commit to this branch once I've checked the other failure aside from the read replica one.